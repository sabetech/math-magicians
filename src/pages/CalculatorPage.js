import React from 'react';
import Calculator from '../components/Calculator';

function CalculatorPage() {
  return (
    <div className="calculatorPage">
      <h2>Let&lsquo;s do some Maths</h2>
      <Calculator />
    </div>
  );
}

export default CalculatorPage;
