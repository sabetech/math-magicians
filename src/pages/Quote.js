import React from 'react';

function Quote() {
  return (
    <blockquote>
      Mathematics is not about numbers, equations, computations, or algorithms:
      it is about understanding - William Paul Thurston
    </blockquote>
  );
}

export default Quote;
