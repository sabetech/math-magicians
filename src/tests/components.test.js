import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../pages/Quote';
import CalculatorPage from '../pages/CalculatorPage';

describe('Testing pages that should render correctly', () => {
  it('Quote should render correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Calculator page should render correctly', () => {
    const tree = renderer.create(<CalculatorPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
