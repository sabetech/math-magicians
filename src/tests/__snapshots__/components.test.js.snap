// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cal should render correctly 1`] = `
<blockquote>
  Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding - William Paul Thurston
</blockquote>
`;

exports[`Calulator page should render correctly 1`] = `
<div
  className="calculatorPage"
>
  <h2>
    Let‘s do some Maths
  </h2>
  <div
    className="calculator parent"
  >
    <p
      className="display"
    >
      0
       
       
    </p>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
    <button
      className="calcButton zeroButton"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
  </div>
</div>
`;

exports[`Quote should render correctly 1`] = `
<blockquote>
  Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding - William Paul Thurston
</blockquote>
`;

exports[`Testing pages that should render correctly Calculator page should render correctly 1`] = `
<div
  className="calculatorPage"
>
  <h2>
    Let‘s do some Maths
  </h2>
  <div
    className="calculator parent"
  >
    <p
      className="display"
    >
      0
       
       
    </p>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
    <button
      className="calcButton zeroButton"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
  </div>
</div>
`;

exports[`Testing pages that should render correctly Calulator page should render correctly 1`] = `
<div
  className="calculatorPage"
>
  <h2>
    Let‘s do some Maths
  </h2>
  <div
    className="calculator parent"
  >
    <p
      className="display"
    >
      0
       
       
    </p>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
    <button
      className="calcButton zeroButton"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
  </div>
</div>
`;

exports[`Testing pages that should render correctly Home page should render correctly 1`] = `
<div
  className="calculatorPage"
>
  <h2>
    Let‘s do some Maths
  </h2>
  <div
    className="calculator parent"
  >
    <p
      className="display"
    >
      0
       
       
    </p>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="calcButton "
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="calcButton  operatorButton"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
    <button
      className="calcButton zeroButton"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
  </div>
</div>
`;

exports[`Testing pages that should render correctly Quote should render correctly 1`] = `
<blockquote>
  Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding - William Paul Thurston
</blockquote>
`;
